### DISCUSSION TITLE
milvus dr

### DISCUSSION THREAD

**Discussion by rere950303** [milvus-io/milvus#35445] [Q&A and General discussion] [answered]
When deploying a new milvus cluster by performing minio's own bucket replication(https://min.io/docs/minio/linux/administration/bucket-replication.html) and pointing it at the replication target minio, the collection is not loaded on the querynode. Is this because the metadata of the same shape as the replicated minio is not stored in etcd? I am wondering if I need to use milvus' own backup/restore for milvus dr.

**Comment by yhmo** [comment, marked as answer]
It is not a proper way to backup minio data and etcd meta respectively.
There are asynchronous tasks inside milvus to process compaction, building index, segment persistence, etc. Typically, a task puts meta into etcd once after data is successfully stored into minio. If some tasks are not finished and you backup the bucket, some items might missed in the backup bucket, but their meta is put into etcd. Then when you switch the bucket, the meta and data is not consistent and lead to internal errors. I guess the querynode log contains errors that some data files were not found in the bucket.

The milvus-backup tool is the recommended/official tool to do backup/restore work.

**Reply by rere950303** [reply]
thanks for replying!!
